### ISSUE TITLE
[BUG] Receiving avoid dynamic warning in Map with type <String, dynamic>

### ISSUE THREAD

**Issue by quoc-huynh-cosee** [dart-code-checker/dart-code-metrics#1191]
### Environment and configuration

DCM version: 5.5.1
Dart SDK version: 2.19.2

<details>
<summary>Configuration</summary>

```yaml
dart_code_metrics:
  rules:
    - avoid-dynamic
```
</details>

### What did you do?

```dart
final myMap = <String, dynamic>{};
```

### What did you expect to happen?

No lint warning.

### What actually happened?

I'm receiving the hint **Avoid using dynamic type.**

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by incendial** [comment]
@quoc-huynh-cosee why do you need exactly dynamic here? Have you tried replacing it with `Object?`?

**Comment by quoc-huynh-cosee** [comment]
Yeah it's working with `Object?`, but shouldn't this :
```dart
final Map<String, dynamic> myMap = {}; // No warning
```

behave the same as this:
```dart
final myMap = <String, dynamic>{}; // avoid-dynamic warning
```

**Comment by incendial** [comment]
Available in 5.7.0 🚀
